components/modal/modal-confirm-container.component.ts

```diff
@@ -104,7 +109,19 @@ import { ModalOptions } from './modal-types';
     '(@modalContainer.start)': 'onAnimationStart($event)',
     '(@modalContainer.done)': 'onAnimationDone($event)',
     '(click)': 'onContainerClick($event)'
-  }
+  },
+  imports: [
+    NgClass,
+    NgStyle,
+    NzPipesModule,
+    NzIconModule,
+    NgIf,
+    NzModalCloseComponent,
+    NzOutletModule,
+    PortalModule,
```

Comment:
suggestion: maybe in this case it's not needed to import all the PortalModule just to add the cdkPortalOutlet directive which is standalone. What do you think about importing only what we need to optimize the final bundle ?

Reply:
Yeah i'm on the same page with importing only the required modules, but apparently `cdkPortalOutlet` is not standalone:
![image](https://github.com/NG-ZORRO/ng-zorro-antd/assets/62149413/d7f71579-78fa-4394-b782-d1edc4711660)

Reply:
> Yeah i'm on the same page with importing only the required modules, but apparently `cdkPortalOutlet` is not standalone: ![image](https://private-user-images.githubusercontent.com/62149413/287831342-d7f71579-78fa-4394-b782-d1edc4711660.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3MjM2OTcsIm5iZiI6MTcwMTcyMzM5NywicGF0aCI6Ii82MjE0OTQxMy8yODc4MzEzNDItZDdmNzE1NzktNzhmYS00Mzk0LWI3ODItZDFlZGM0NzExNjYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA0VDIwNTYzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhYjIwOWQ0MzQ5MGYwODdmNGMxNDNmMmUwYzgyMWM3N2UzMGM5ODU5N2U0NWY2YjE1MTMxYWRiZDgxM2E0Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.blGbCreYluI9in3kyU7Brnkic5j-lRZ15wIo-CVaEjY)

Mmmh; if you take a look at this link you can see that the directive is mark as standalone: 

https://github.com/angular/components/blob/main/src/cdk/portal/portal-directives.ts

Reply:
Yeah you are right, but it was added to version 17.1.0-next.1.

https://github.com/angular/components/blame/0762d69b394343555488b5aaada9444dd1a5083a/src/cdk/portal/portal-directives.ts#L74

Reply:
we are currently using `@angular/cdk": "^17.0.0`

Reply:
> Yeah you are right, but it was added to version 17.1.0-next.1.
> 
> https://github.com/angular/components/blame/0762d69b394343555488b5aaada9444dd1a5083a/src/cdk/portal/portal-directives.ts#L74

Ohh yeah true miss it !! Thanks. 

We will update this after. 

So LGTM

thanks for your work !!